README.md

```diff
@@ -0,0 +1,83 @@
+This module interfaces with the TLS implementation of Oracle JDK8 to
+provide an implementation of ALPN.  The module was written for use in
+Grizzly, but is intended as a general purpose solution to allow JDK8
+based software to use ALPN.
+
+---
+
+## How to Avoid IP Taint When Examining this Code
+
+Note that everything in the `api` maven sub-module of this module is
+safe to look it without fear of
+[intellectual property taint](http://thekroliks.typepad.com/whereas/2009/03/inlicensing-technology-3-things-you-need-to-avoid-tainting.html).
+It is all interfaces and one utility POJO.
+
+---
+
+# How Grizzly Uses This Module
+
+Grizzly has the concept of an
+[`AddOn`](https://github.com/javaee/grizzly/blob/e0e9200479851078d9cf2bad1cf29fa72f525437/modules/http-server/src/main/java/org/glassfish/grizzly/http/server/AddOn.java).
+This facility allows Grizzly to be extended using an implementation of
+the Chain of Responsibility design pattern.  In general, an `AddOn`
+implementation will insert one or more `Filter` implementations into the
+`FilterChain` which is used to process HTTP requests and cause HTTP
+responses to be sent.
+
+Grizzly itself uses this `AddOn` concept to provide HTTP/2 support, in
+the form of
+[`Http2AddOn`](https://github.com/javaee/grizzly/blob/e0e9200479851078d9cf2bad1cf29fa72f525437/modules/http2/src/main/java/org/glassfish/grizzly/http2/Http2AddOn.java)
+This `AddOn` implementation inserts some filters in the chain in the
+proper order to ensure that ALPN concerns are handled first, and then
+the HTTP/2 protocol is handled.  The former is of interest here.
```

Comment:
Better phrased as something like:  in addition to registering filters for HTTP/2 processing, it registers a callback with the ALPN extension.

Reply:
Done.